pkg/async/pool/pool.go

```diff
@@ -121,11 +121,12 @@ type Options struct {
 
 // Pool structure
 type Pool struct {
-	queue   chan unit
-	opts    *Options
-	wg      *sync.WaitGroup
-	cancel  func(error)
-	context context.Context
+	contextMu sync.Mutex
```

Comment:
it would be good to have a comment on exactly what the mutex is protecting, or if it's just protecting the entire struct.  contextMu to me implies it's protecting the context, but I see it used outside of just protecting the context.

Reply:
Let me leave a comment about that. It actually is just protecting the context because things like `orerr.CancelWithError` modify the context.

Let me also narrow down the locking a bit to make that more clear.

Reply:
Updated. That should clear things up a bit.

Reply:
ah i see now.  cancel is the method pulled from withcancel, so it's all attributes of the context.  as a tiny nit, i would organize the struct vars like that. but w/e.  thjx!

i.e.
```
contextMu
context
cancel

opts
queue
wg
```